AdobeStockAsset/Test/Integration/Model/CategoryRepositoryTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeStockAsset\Test\Integration\Model;
+
+use Magento\AdobeStockAssetApi\Api\CategoryRepositoryInterface;
+use Magento\AdobeStockAssetApi\Api\Data\AssetSearchResultsInterface;
+use Magento\AdobeStockAssetApi\Api\Data\CategoryInterface;
+use Magento\AdobeStockAssetApi\Api\Data\CategoryInterfaceFactory;
+use Magento\Framework\Api\FilterBuilder;
+use Magento\Framework\Api\Search\FilterGroupBuilder;
+use Magento\Framework\Api\Search\SearchCriteriaBuilder;
+use Magento\Framework\Api\SortOrder;
+use Magento\Framework\Api\SortOrderBuilder;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\TestFramework\Helper\Bootstrap;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Provide integration tests for the Adobe Stock CategoryRepository functionality.
+ */
+class CategoryRepositoryTest extends TestCase
+{
+    private const FIXTURE_ASSET_CATEGORY_ID = 42;
+
+    /**
+     * @var CategoryRepositoryInterface
+     */
+    private $categoryRepository;
+
+    /**
+     * @inheritdoc
+     */
+    public function setUp()
+    {
+        $this->categoryRepository = Bootstrap::getObjectManager()->get(CategoryRepositoryInterface::class);
+    }
+
+    /**
+     * Test getting an Adobe Stock category by id.
+     *
+     * @magentoDataFixture ../../../../app/code/Magento/AdobeStockAsset/Test/_files/category.php
+     */
+    public function testGetById(): void
+    {
+        /** @var CategoryInterface $expectedCategory */
+        $expectedCategory = $this->categoryRepository->getById(self::FIXTURE_ASSET_CATEGORY_ID);
+
+        $this->assertEquals($expectedCategory->getId(), self::FIXTURE_ASSET_CATEGORY_ID);
+    }
+
+    /**
+     * Test delete an Adobe Stock category by id.
+     *
+     * @magentoDataFixture ../../../../app/code/Magento/AdobeStockAsset/Test/_files/category.php
+     */
+    public function testDeleteById(): void
+    {
+        $this->categoryRepository->deleteById(self::FIXTURE_ASSET_CATEGORY_ID);
+        try {
+            $this->categoryRepository->getById(self::FIXTURE_ASSET_CATEGORY_ID);
+            $this->fail('Expected NoSuchEntityException caught');
+        } catch (NoSuchEntityException $exception) {
+            $message = __(
```

Comment:
Please do not verify the exception message, exception type should be enough. Also, please use expectedException phpunit functionality